batch-oai-harvest/src/main/java/data/config/OaiHarvestJobConfig.java

```diff
@@ -27,46 +32,71 @@
+                .build();
+    }
+
+    @Bean("oaiIdentifiersHarvestStep")
+    public Step oaiRepositoryIdentifiersHarvestStep(
+            OaiIdentifiersEndpointItemReader oaiIdentifiersEndpointItemReader,
+            OaiIdentifiersWriter oaiIdentifiersWriter,
+            JobRepository jobRepository,
+            @Qualifier("transactionManager") PlatformTransactionManager transactionManager,
+            @Qualifier("oaiHarvestStepAsyncTaskExecutor") TaskExecutor oaiHarvestStepAsyncTaskExecutor) {
 
-  @Value("${oaiharvest.chunk.size}")
-  public int chunkSize;
-  @Value("${oaiharvest.parallelization.size}")
-  public int parallelization;
+        return new StepBuilder(IDENTIFIERS_HARVEST_STEP_NAME, jobRepository)
+                .<ExecutionRecordExternalIdentifier, ExecutionRecordExternalIdentifier>chunk(chunkSize, transactionManager)
+                .reader(oaiIdentifiersEndpointItemReader)
+                .writer(oaiIdentifiersWriter)
+//                .taskExecutor(oaiHarvestStepAsyncTaskExecutor)
+                .build();
+    }
 
-  @Bean
-  public Job oaiHarvestBatchJob(JobRepository jobRepository, Step oaiHarvestStep) {
-    LOGGER.info("Chunk size: {}, Parallelization size: {}", chunkSize, parallelization);
-    return new JobBuilder(BATCH_JOB, jobRepository)
-        .incrementer(new TimestampJobParametersIncrementer())
-        .start(oaiHarvestStep)
-        .build();
-  }
+    @Bean("oaiRecordsHarvestStep")
+    public Step oaiRecordsHarvestStep(
+            JobRepository jobRepository,
+            OaiIdentifiersRepositoryItemReader oaiIdentifiersReader,
+            @Qualifier("transactionManager") PlatformTransactionManager transactionManager,
+            OaiRecordHarvester oaiRecordHarvester,
+            ExecutionRecordDTOItemWriter writer,
+            @Qualifier("oaiHarvestStepAsyncTaskExecutor") TaskExecutor oaiHarvestStepAsyncTaskExecutor) {
 
-  @Bean
-  public Step oaiHarvestStep(JobRepository jobRepository,
-      @Qualifier("transactionManager") PlatformTransactionManager transactionManager,
-      OaiHarvestItemReader oaiHarvestItemReader,
-      ExecutionRecordDTOItemWriter writer,
-      @Qualifier("oaiHarvestStepAsyncTaskExecutor") TaskExecutor oaiHarvestStepAsyncTaskExecutor) {
-    return new StepBuilder(STEP_NAME, jobRepository)
-        .<ExecutionRecordDTO, ExecutionRecordDTO>chunk(chunkSize, transactionManager)
-        .reader(oaiHarvestItemReader)
-        .writer(writer)
-        //TODO: 2024-01-31 - Update with a better parallelization reader and remove Step parallization.
-        .taskExecutor(oaiHarvestStepAsyncTaskExecutor)
-        .throttleLimit(parallelization)
-        .build();
-  }
+        return new StepBuilder(RECORDS_HARVEST_STEP_NAME, jobRepository)
+                .<ExecutionRecordExternalIdentifier, ExecutionRecordDTO>chunk(chunkSize, transactionManager)
+                .reader(oaiIdentifiersReader)
+                .processor(oaiRecordHarvester)
```

Comment:
I just realized you haven't used the Async objects for the processor and writer.
I believe those are necessary to actually have parallelism.

Reply:
done